app/questionnaire/navigator.py

```diff
@@ -101,6 +101,11 @@ def __init__(self, survey_json, metadata=None, answer_store=None):
         self.metadata = metadata or {}
         self.survey_json = survey_json
 
+        self.preceeding_path = []
```

Comment:
Mmm, yes - I've thought this in the past. An introduction is unique from a block in that its structure is different and the content is rendered using a different template.